developer-guide.md

```diff
@@ -0,0 +1,545 @@
+  "properties": {
+    "text": {
+      "type": "dense_vector",
+      "dims": 8
+      "similarity": "cosine"
+      "index": True
+    }
+  }
+})
+```
+
+To check if the index was created, you can run the following command:
+
+```python
+es.indices.get(index="demo")
+```
+
+### Index Documents
+
+Lets start by ingesting vectors into your index, using the bulk helper.
+
+```python
+from elasticsearch.helpers import bulk
+
+docs = [
+  { "text": [0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5], "title": "Document 1"},
+  { "text": [0.4, 0.4, 0.4, 0.4, 0.4, 0.4, 0.4, 0.4], "title": "Document 2"},
+  { "text": [0.3, 0.3, 0.3, 0.3, 0.3, 0.3, 0.3, 0.3], "title": "Document 3"}
+]
+
+actions = []
+for doc in docs:
+  action = {
+      "_index": "demo",
+      "_source": doc
+  }
+  actions.append(action)
+
+bulk(es, actions, refresh=True)
+
+```
+
+We use `refresh` to make sure the documents are available for search immediately.
+
+### Search for Similar Documents
+
+Now that we have indexed some documents, we can search for similar documents that are most similar to the query vector.
+
+We will calculate the distance with the cosine similarity function, and return the top 10 most similar documents.
+
+```python
+response = es.search(index="demo", body={
+    "knn": {
+      "field": "text",
+      "query_vector": [0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5],
+      "k": 10,
+      "num_candidates": 100
+    }
+})
+
+```
+
+# Setup
+
+Take you through the steps to setup and connect to Elasticsearch.
+
+## Running Elasticsearch
+
+You can start using Elasticsearch really quickly. We highlight two of the most common ways to use Elasticsearch below.
+
+### Example: Use Elasticsearch with Docker
+
+Recommended for local development and testing.
+
+The quickest way to get started developing with Elasticsearch is to use Docker. Run the following command to download and launch the official Elasticsearch Docker image:
+
+```bash
+docker run -p 9200:9200 \
+  -e "discovery.type=single-node" \
+  -e "xpack.security.enabled=false" \
+  -e "xpack.security.http.ssl.enabled=false" \
+  -e "run.license_type=trial" \
+  docker.elastic.co/elasticsearch/elasticsearch:8.9.0
+```
+
+Elasticsearch is now running on http://localhost:9200
+
+### Example: Use Elasticsearch with Elastic Cloud
+
+Recommended for production.
+
+Elastic Cloud is the official hosted and managed Elasticsearch service, offering Elasticsearch, Kibana, APM, and much more. Elastic Cloud is available on AWS, GCP, and Azure.
+
+You can sign up for a free 14-day trial at https://cloud.elastic.co
+
+## API Clients
+
+Elasticsearch supports a number of REST clients for different languages.
+
+- Python [view docs](https://elasticsearch-py.readthedocs.io/en/master/)
+- Javascript [view docs](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/getting-started-js.html)
+
+For other clients, see [Client Libraries](https://www.elastic.co/guide/en/elasticsearch/client/index.html).
+
```

Comment:
Maybe just add a note saying that we'll be using Python exclusively in this guide?

Reply:
Yeh - when this guide goes onto search labs, it wont just be python. I dont think that line is needed.